Add MetOffice weather forecast and daily update

MrHarcombe opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->


## Proposed change
<!-- 
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Adding forecast data to the Met Office weather integration, as well as daily update modes.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [X] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example configuration.yaml
weather:
  -
    platform: metoffice
    api_key: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
    name: Hourly
    mode: 3hourly
  -
    platform: metoffice
    api_key: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
    name: Daily
    mode: daily
```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: N/A
- This PR is related to issue: N/A
- Link to documentation pull request: (incoming)

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [X] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [x] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

balloob committed:
Cloud do checks during setup (#33507)

* Update cloud to do more tasks during async_setup

* Upgrade hass_nabucasa to 0.33

janiversen committed:
Fix modbus sync/async issues (#34043)

* add pyserial to manifest

pymodbus is very developer oriented and assumes every developer
adapt the requierements.txt to his/hers needs.

Our requirements.txt is different it contains all posibilities allowing
user to later change configuration without having to install extra
packages.

As a consequence manifest.json needs to include the pyserial.

* modbus: make truly async client creation

Make hass call listen_once async.
Integrate content of start_modbus into async_setup.

Do not use the boiler plate create tcp client function from
pymodbus as it is sync, and also does not work well with asyncio,
instead call the init_<type> directly, since that is async.

* both component/modbus and component/serial uses pyserial-async
but with slighty different version requirements.

Combined the 2.

* Review 1

* Review 2

* Review

@staticmethod is no good, because the function uses class variables.

* Review

Pytest is sometimes a bit cryptic, lets hope this does it.

frenck committed:
Temporary transition Docker init (#34135)

ctalkington committed:
Catch IPPVersionNotSupportedError in IPP (#34184)

* Update config_flow.py

* squash.

* Update test_config_flow.py

* Update config_flow.py

* Update test_config_flow.py

* Update test_config_flow.py

* Update test_config_flow.py

* Update test_config_flow.py

* Update test_config_flow.py

bachya committed:
Fix websocket connection bug/errant logic in Ambient PWS (#34217)

fredrike committed:
Add daikin update_before_add (#34248)

fredrike committed:
Fix various Daikin issues (#34249)

* various Daikin fixes

* make timeout a constant

timmo001 committed:
Add AsusWRT Devices Connected Sensor (#34204)

* Add Devices Connected

* Remove attributes

* Add sensors to test

* Improve sensor test

* Cleanup

* Apply suggestions from code review

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Import device from aioasuswrt

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

balloob committed:
Fix Cloud UI bug preventing managing Google 2FA (#34241)

* Fix Cloud UI bug preventing managing Google 2FA

* Update comment

balloob committed:
Fix Cloud UI bug preventing managing Google 2FA (#34241)

* Fix Cloud UI bug preventing managing Google 2FA

* Update comment

balloob committed:
Bumped version to 0.108.5

balloob committed:
Move title translation to root (#33850)

balloob committed:
Merge pull request #34260 from home-assistant/rc

0.108.5

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

balloob committed:
Update translations

balloob committed:
Fix translations

presslab-us committed:
Add MQTT climate temperature unit (#34066)

* Add MQTT temperature unit

balloob committed:
Add command to get integration manifests (#34262)

* Add command to get integration manifests

* Add is_built_in value to manifest

* Update APIs

balloob committed:
Add Translations 2.0 migrate script (#34261)

ziv1234 committed:
Removed uncaught exceptions from Dyson (#34112)

* fixed what seems to be a typo

* added load_mock_device in common.py so it loads all the required things into the mocks
so they don't throw exceptions for mocks not being able to convert to int

* reverted change in homeassistant/components/dyson/sensor.py
added both values to the mock device (volatil and volatile)

balloob committed:
Patch http.client to not do I/O in the event loop (#34194)

balloob committed:
Add translations check script (#34272)

homeassistant committed:
[ci skip] Translation update

dannerph committed:
Fix Keba request data service call (#34254)

ctalkington committed:
Support DirecTV music channels with extended meta (#34228)

bdraco committed:
Add tests for additional nut ups models (#34240)

* Add NUT tests for an Eaton 5E650I UPS

* add a test to make sure multiple ups get setup

* Add a few more upses

* fix missing status key

* cover

* Force CI run

bdraco committed:
Set homekit alarm/sensor/switch/cover state as soon as possible (#34245)

* Set homekit alarm/sensor/switch state as soon as possible

This change is part of a multi-part effort to fix the
HomeKit event storms on startup.

Previously we would set the states after HomeKit
had started up which meant that when the controller
client connected it would request the states and get
a list of default states so all the initial states
would always be wrong. The defaults states generally went
unnoticed because we set the state of each HomeKit device
soon after which would result in an event storm in the log
that looked like the following for every client and every
device:

Sending event to client: ('192.168.x.x', 58410)
Sending event to client: ('192.168.x.x', 53399)
Sending event to client: ('192.168.x.x', 53399)

To solve this, we now set the state right away when we
create the entity in HomeKit, so it is correct on
initial sync, which avoids the event storm.  Additionally,
we now check all states values before sending an update
to HomeKit to ensure we do not send events when nothing
has changed.

* pylint

* Fix event storm in covers as well

* fix refactoring error in security system

* cover positions, now with constants

bdraco committed:
Convert homekit fans to use service callbacks (#34229)

* Convert homekit fans to use service callbacks

* Convert homekit fans to use service callbacks

Service callbacks allow us ensure that we call
fan services in the correct order.

* Avoid calling turn_on if a speed is sent and the device is on

* Fix test to not leave files behind

* Fix test

* Update homeassistant/components/homekit/type_fans.py

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

bdraco committed:
Prevent a single accessory setup failure from breaking all HomeKit accessories (#34263)

* Prevent a single accessory setup failure from breaking all HomeKit accessories

Raise the max devices to 150 as the spec allows for this
many.  Previously 100 made sense because of the event
storms when homekit started would sometimes break pairing,
as these have largely been fixed in 0.109 (still a few
to cleanup) using the HAP spec limit of 150 is now possible.

* Handle both failure states

bdraco committed:
Fix synology_dsm i/o in event loop (#34281)

leppa committed:
Improve MAX! Cube integration (#28845)

* maxcube: Make it possible to return to auto mode

It's currently not possible to switch back to auto (schedule) mode after
changing to manual or vacation. Add `PRESET_NONE` to list of presets
that will return thermostat back into auto (schedule) mode.

Additionally, use `PRESET_AWAY` instead of custom preset name for
vacation mode.

* maxcube: Provide `hvac_action` based on valve state

Not supported for wall thermostats.

* maxcube: Add support for Comfort, Eco, Off and On modes

Off is mapped to HVAC_OFF, while On - to HVAC_HEAT.

* maxcube: Add `hvac_action` support for wall thermostats

We check all thermostats in the same room as the wall thermostat. If at
least one of them has its valve open - the room is being heated.

* maxcube: Expose valve position as state attribute

Also fix a small logical error in `hvac_action`.

* maxcube: Fix linter errors and formatting

* maxcube: Adjust mapping between MAX! and HA modes and presets

MAX! 'Manual' mode now corresponds to 'Heating' mode of HA. MAX! 'Eco'
and 'Comfort' temperatures are 'Heating' mode presets. MAX! 'On' mode is
now a 'Heating' preset, too.

* maxcube: Address review comments

Ernst79 committed:
Add state to RFXtrx covers (#30935)

* Add state to rfxtrx cover

* Add state to rfxtrx cover (cover.py)

MatthewFlamm committed:
Add config flow to nws and remove yaml configuration (#34267)

* add config flow to nws and remove yaml

* Don't duplicate scan_time

Co-Authored-By: J. Nick Koston <nick@koston.org>

* Use _abort_if_unique_id_configured

Co-Authored-By: J. Nick Koston <nick@koston.org>

* fix abort

* Add unavailable tests

* update and use better strings

* lint

Co-authored-by: J. Nick Koston <nick@koston.org>

lewei50 committed:
Bump iammeter to 0.1.7 to fix empty SN (#34279)

* update pypi lib to fix empty SN issue.

* Update requirements_all.txt

vanbalken committed:
Add device tracking support for the Arris TG2492LG router (#30972)

* Add device tracker based on arris-tg2492lg

* Return name of device and update version of arris_tg2492lg

* Update CODEOWNERS, requirements_all.txt and .coveragerc

* Change default url to host

* Use f-strings

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

* Clean up

* Fix formatting

Co-authored-by: Franck Nijhof <frenck@frenck.nl>
Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

balloob committed:
Take integration title from manifest if not translated (#34283)

balloob committed:
Allow hassfest to validate specific integrations (#34277)

amelchio committed:
Support contemporary entity selectors for LIFX services (#33062)

* Support contemporary entity selectors for LIFX services

* Use async_register_entity_service() for lifx.set_state

* Call directly

kit-klein committed:
Ensure zone in update payload for konnected (#34289)

Danielhiversen committed:
Upgrade broadlink lib to 0.13.1 (#34290)

Danielhiversen committed:
Upgrade broadlink lib to 0.13.1 (#34290)

balloob committed:
Migrate frontend translations of domains to backend (#34294)

balloob committed:
Drop title from translations if brand name (#34306)

balloob committed:
Update translations

kit-klein committed:
Ensure konnected unsubscribes during entry unloads (#34291)

ctalkington committed:
Improve Sonarr Upcoming Time Handling (#34224)

ctalkington committed:
Improve IPP Config Flow (#34212)

cyr-ius committed:
Add Friends of Hue Switch - Model FOHSWITCH (ZGPSWITCH) (#34195)

raman325 committed:
Refactor vizio media_player tests to remove conditional statements from helper function (#33615)

raman325 committed:
Skip ignored hosts when checking existing config entries in config flow (#34280)

raman325 committed:
Skip ignored hosts when checking existing config entries in config flow (#34280)

basnijholt committed:
Pass an argument to kef.update_dsp for async_track_time_interval (#34310)

basnijholt committed:
Pass an argument to kef.update_dsp for async_track_time_interval (#34310)

bdraco committed:
Use config entry id for unique id if serial number is missing (#34154)

bdraco committed:
Use config entry id for unique id if serial number is missing (#34154)

Kane610 committed:
UniFi - Support automatic removal of clients (#34307)

amelchio committed:
Purge recorder database at night (#33646)

amelchio committed:
Fixes for Sonos media titles (#34311)

amelchio committed:
Fixes for Sonos media titles (#34311)

bdraco committed:
Fix default elkm1 temp units (#34274)

bdraco committed:
Fix default elkm1 temp units (#34274)

bdraco committed:
Resolve homekit not updating motion sensors (#34282)

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

raman325 committed:
Abort vizio zeroconf config flow if unique ID is already configured (#34313)

raman325 committed:
Abort vizio zeroconf config flow if unique ID is already configured (#34313)

balloob committed:
Update Coordinator: Only schedule a refresh if listenerrs (#34317)

balloob committed:
Update Coordinator: Only schedule a refresh if listenerrs (#34317)

danielperna84 committed:
Update pyhomematic to 0.1.66 (#34314)

homeassistant committed:
[ci skip] Translation update

jnimmo committed:
Bump to pyIntesisHome 1.7.4 (#34319)

bdraco committed:
Restore isy light brightness after off (#34320)

balloob committed:
Bumped version to 0.108.6

balloob committed:
Improve error message when people have not moved config flow title yet (#34321)

balloob committed:
Merge pull request #34323 from home-assistant/rc

Kane610 committed:
UniFi - Allow tracking of clients connected to third party APs (#34067)

* Allow disable wired bug work around

* Move small improvements from closed PR #34065

* Fix failing test

* Add new test

* Some extra logging

* Harmonize log outputs

* Add config flow string

* Fix Balloobs comments

balloob committed:
Remove two more titles from strings.json (#34324)

janiversen committed:
Rollback modbus to version 0.107.7 keep new functionality (#34287)

* Rollback modbus to version 0.107.7

Update manifest to not use async.

Rollback entities to sync version.

Keep newer modifications apart from async.

Rollback __init__ to sync version but keep the new functionality.

add async sub directory

Adding the current (not working) version in a sub directory,
to allow easy sharing with a few alfa testers.

The async version are to be updated to use the serial/tcp already
available instead of the flaky pymodbus version. pymodbus is still
needed to encode/decode the messagess.

Update test cases to reflect sync implementation, but
keep the new functionality like e.g. conftest.py.

* do not publish async version

The async version will be made available in a forked repo, until
it is ready to replace the production code.

janiversen committed:
Rollback modbus to version 0.107.7 keep new functionality (#34287)

* Rollback modbus to version 0.107.7

Update manifest to not use async.

Rollback entities to sync version.

Keep newer modifications apart from async.

Rollback __init__ to sync version but keep the new functionality.

add async sub directory

Adding the current (not working) version in a sub directory,
to allow easy sharing with a few alfa testers.

The async version are to be updated to use the serial/tcp already
available instead of the flaky pymodbus version. pymodbus is still
needed to encode/decode the messagess.

Update test cases to reflect sync implementation, but
keep the new functionality like e.g. conftest.py.

* do not publish async version

The async version will be made available in a forked repo, until
it is ready to replace the production code.

ziv1234 committed:
Fix uncaught exceptions in ios (#34119)

* verify that the config in hass is not empty

* changed to use MockConfigEntry

* Update tests/components/ios/test_init.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/ios/test_init.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* changed the test per suggestions

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

ziv1234 committed:
Fix uncaught exception in local_file (#34312)

* remove ignore from localfile

* added mock_registry

* since there are no more exceptions, infra to ignore them is redundant

frenck committed:
Upgrade yamllint to 1.23.0 (#34337)

tubalainen committed:
Add rflink binary_sensor allon and alloff commands (#32411)

* added "allon" and "alloff" to the binary_sensor

Ref to the issue: https://github.com/home-assistant/core/issues/32399

* Cosmetic fix

* Fix format

* Update test

Co-authored-by: Erik Montnemery <erik@montnemery.com>

bdraco committed:
Ensure nexia state file is in a writable location (#34325)

* bump nexia to 0.9.2

bdraco committed:
Ensure nexia state file is in a writable location (#34325)

* bump nexia to 0.9.2

balloob committed:
Add a script to clean the frontend translations (#34309)

azogue committed:
Fix missing events for hue remotes (#34340)

bdraco committed:
Fix emulated_hue brightness off by one (#34185)

* Fix emulated_hue brightness off by one

Hue uses a max brightness of 254, Home Assistant
uses a max brightness of 255. Values > 127 were
off by one.

* use constant

* fix test

* add debug

* Revert "add debug"

This reverts commit 242220a02e8d8e9ec5e21930470b2adb97c75e2b.

hobbe committed:
Add trunk and frunk locks to Tesla integration (#34343)

emontnemery committed:
Bump zeroconf to 0.25.1 (#34341)

scop committed:
Type hint improvements (#33082)

Commit:
Fix HomematicIP smoke detector detection type (#34347)

dfournie committed:
Add support for DEXXO SMART io Garage Opener (#27704)

* Add support for DEXXO SMART io Garage Opener

* Add support for DEXXO SMART io Garage Opener

* Update with with latest release 0.108.5

* Update with release 0.108.5

* Remove empty lines

* Fix code formatting

jrester committed:
Update tesla-powerwall to version 0.2.5 (#34348)

bdraco committed:
Add missing services to Homekit services yaml (#34349)

* Add missing services to homekit services.yaml

* Add missing services to homekit services.yaml

MartinHjelmare committed:
Disable pylint invalid name for TypeVar T (#34355)

Quentame committed:
Add missed CONF_API_VERSION config in flow + Fix missing_data translation (#34356)

* Add missed CONF_API_VERSION config in flow + remove CONF_NAME in tests

* Fix missing_data translation

jrester committed:
Use serial numbers for unique_id of powerwall devices (#34351)

* Update tesla-powerwall to version 0.2.5

* Create unique_id from serial numbers of powerwalls
	modified:   homeassistant/components/powerwall/__init__.py
	modified:   homeassistant/components/powerwall/binary_sensor.py
	modified:   homeassistant/components/powerwall/const.py
	modified:   homeassistant/components/powerwall/entity.py
	modified:   homeassistant/components/powerwall/sensor.py
	modified:   tests/components/powerwall/mocks.py
	modified:   tests/components/powerwall/test_sensor.py

* Fix pylint error
	modified:   homeassistant/components/powerwall/__init__.py

Kane610 committed:
deCONZ - Always allow manual input of gateway (#33951)

* Always allow manual input of gateway

Quentame committed:
Add 2SA authentication to Synology DSM (#34101)

Co-Authored-By: J. Nick Koston <nick@koston.org>

Co-authored-by: J. Nick Koston <nick@koston.org>

homeassistant committed:
[ci skip] Translation update

FredericMa committed:
Extend knx brightness with rgb brightness if brightness addres… (#33152)

* Extend knx brightness with rgb brightness if brightness addresses are not supported

* Fix explicit return value

* Check for None

* Remove not needed check

* Disable false positive pylint warning

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

jjlawren committed:
Ignore non-ASCII keys in zeroconf payloads (#34344)

* Ignore non-ASCII keys in zeroconf payloads

* Don't rely on logging for test

chrisrosset committed:
Use name instead of friendly name in uscis config (#33431)

The USCIS integration's code and documentation use different properties
for the sensor name. The code uses `friendly_name` and the docs state `name`.

https://github.com/home-assistant/home-assistant.io/pull/12524#issuecomment-605604002
suggested to make the changes in the code as `friendly_name` is not usually used
in this way.

Commit:
Add config flow to panasonic_viera component (#33829)

* Updating the panasonic_viera component

* Updating .coveragerc

* Removing testplatform

* Updating strings.json

* Commit before rebase

* Commit before rebase

* Commit before rebase

* Commit before rebase

* Commit before rebase

* Adding tests and stuff

* Fixing permission issues

* Ignoring Pylint warnings

* Fixing one more Pylint warning

* Refactoring

* Commiting changes - part 1

* Commiting changes: part 2

* Turning unknown error logs into exception logs

* Update strings.json

* Rebasing

* Updating the panasonic_viera component

* Removing testplatform

* Updating strings.json

* Commit before rebase

* Commit before rebase

* Commit before rebase

* Commit before rebase

* Commit before rebase

* Adding tests and stuff

* Fixing permission issues

* Ignoring Pylint warnings

* Fixing one more Pylint warning

* Refactoring

* Commiting changes - part 1

* Commiting changes: part 2

* Turning unknown error logs into exception logs

* Adding pt-BR translation

* Removing Brazilian Portugues translations

* Modifying error handling

* Adding SOAPError to except handling

* Updating translation

* Refactoring async_step_import

* Fixing indentation

* Fixing requirements after rebase

* Fixing translations

* Fixing issues after rebase

* Routing import step to user step

* Adding myself as a codeowner

bramkragten committed:
Google local sdk: Set right port, correct device id and add base url (#34358)

MatthewFlamm committed:
Use DataUpdateCoordinator in NWS (#34372)

antonverburg committed:
Pi4ioe5v9xxxx from ToggleEntity to SwitchDevice (#34369)

FredericMa committed:
Allow configuring KNX preset_modes via the operation_modes variable (#33068)

bdraco committed:
Switch harmony to async_on_remove (#34378)

ocalvo committed:
Fix a bug where long sms messages get cut off (#34359)

* Fix a bug where long sms messages get cut off

* Clean up

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

JeffLIrion committed:
Bump androidtv to 0.0.41 (#34382)

ctalkington committed:
Add tests for Roku (#34380)

shenxn committed:
Add Braava support to iRobot Roomba component (#33616)

* Add Braava support to iRobot Roomba component

* Replace async_add_job with async_add_executor_job in roomba

* Improve readability in roomba

* Improve error handling in roomba

* Cleanup async_update in roomba

* Split into multiple files in roomba

* Hide protocal details in braava

* Switch to push in braava

* Bump roombapy version to 1.5.1

* Add roomba files to .coveragerc

* Fix typo

* Remove side effects from init in roomba

* Implement StateVacuumDevice in Roomba

* Add IRobotEntity base class to braava

* Fix state in roomba

* Add @shenxn as a codeowner of braava

bdraco committed:
Fix Tado signal collisons (#34118)

* Fix signal collision in tado with multiple accounts

Tado signals were not unique enough to prevent collisions
between accounts with the same zone ids.

* Bump upstream to fix multiple tado accounts

* switch signal undo to async_on_remove

* Bump version now that upstream is released

* bump python-tado

* While its only the dep that changed since the last run, best to force another CI to be sure

felipediel committed:
Add support to the new Broadlink RM Mini 3 and RM4 Series (#32523)

* Add device type

* Use snake_case for devtype

Co-Authored-By: springstan <46536646+springstan@users.noreply.github.com>

* Validate device type as positive int

* Add device type 0x5f36 to switch

* Use default type for sensors

* Add RM4 to switch platform

* Use snake_case for devtype

* Support multiple types of remote

* Validate ip address

* Improve code readability

* Add const.py to .coveragerc

* Use None for unknown device types

* Fix sensors and standardize platform schemas

* Fix if statement

Co-authored-by: springstan <46536646+springstan@users.noreply.github.com>

homeassistant committed:
[ci skip] Translation update

balloob committed:
Allow fetching translations by categories (#34329)

balloob committed:
Bump frontend to 20200418 (#34407)

jjlawren committed:
Bump plexapi to 3.4.0, remove workarounds (#34395)

lufton committed:
Add new languages to Google Cloud TTS (#34334)

* New languages

Arabic, Indian, Thailand, Mandarin

* Reorder language codes

turbokongen committed:
Access Zoneminder config correctly (#34388)

* use correct conf. Always use .get

* remove .get

bdraco committed:
Add support for turning on swing mode in tado (#34404)

shenxn committed:
Code fixes and cleanup for roomba integration (#34409)

* Remove side effects from properties

* Remove redundant return value in async_send_command

* Move callback registeration to async_added_to_hass

* Code cleanup for roomba

gladhorn committed:
Bump nad_receiver version (#34400)

nad_receiver received a few fixes in 0.0.12, which enables the integration
for me. I assume it may depend on the receiver model, but my receiver only reports
mute state and volume when it's on, while the library used to assume to always
get a reply. With 0.0.11 the nad integration simply excepts on startup,
after bumping the version up it works.

Fixes this:
2020-04-18 22:59:38 ERROR (MainThread) [homeassistant.components.media_player] nad: Error on device update!
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 324, in _async_add_entity
    await entity.async_device_update(warning=False)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 476, in async_device_update
    await self.hass.async_add_executor_job(self.update)
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/nad/media_player.py", line 195, in update
    if self._nad_receiver.main_mute("?") == "Off":
  File "/usr/local/lib/python3.7/site-packages/nad_receiver/__init__.py", line 71, in main_mute
    return self.exec_command('main', 'mute', operator, value)
  File "/usr/local/lib/python3.7/site-packages/nad_receiver/__init__.py", line 62, in exec_command
    return msg.decode().strip().split('=')[1]
IndexError: list index out of range

bdraco committed:
Eliminate homekit media_player event storms on startup (#34399)

This is the final event storm reduction as a followup
to #34245

ctalkington committed:
Improve Roku media player test coverage (#34403)

Danielhiversen committed:
upgrade Tibber lib (#34429)

scop committed:
Fix hassfest validate invocation in Tox (#34428)

Jc2k committed:
Store HomeKit generated accessory id against unique_id where possible (#33109)

* HomeKit: Store generated aid against unique_id where possible

* Fix conflict

* Fix max accessories check

* homekit counts the bridge as an accessory

* Add coverage for aidmanager

* prepare for merge

Co-authored-by: J. Nick Koston <nick@koston.org>

Kane610 committed:
UniFi - refactor entity management (#34367)

* Move removal of sensor entities into a base class

* Fix martins comments on sensors

* Reflect sensor changes on device_tracker platform

* Reflect sensor changes on switch platform

* Improve layering

* Make sure to clean up entity and device registry when removing entities

* Fix martins comments

balloob committed:
Fix translations merging (#34417)

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

clau-bucur committed:
Fix serial integration connection stability (#33067)

* Fix serial integration connection stability

Retry to connect if serial device is not available on start, or connection is lost.
Added some more log entries upon connection or when errors are encountered.

* Updated code based on review

* Fixed exception indentation

* Handle SerialException

* Sort imports

* Update homeassistant/components/serial/sensor.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/serial/sensor.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* More clean up

* Format and delint

* Extract error handler

* Clean up stop handler

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bdraco committed:
Resolve delay sending multiple commands to harmony remotes (#34410)

* Switch harmony to entity service

* Fix delay between sending commands, collapse function

balloob committed:
Fix circular import (#34441)

balloob committed:
Prevent Synology Camera doing I/O in event loop (#34442)

bdraco committed:
Optimize async_generate_entity_id (#34440)

homeassistant committed:
[ci skip] Translation update

bachya committed:
Ensure existing SimpliSafe notifications trigger event on HASS startup (#34232)

* Ensure existing SimpliSafe notifications trigger event on HASS startup

* Don't redefine a standing function each loop iteration

* Wait until HASS state is "running"

* Linting

* Linting

* Code review

* Code review

* Simplify

* Code review

* Cleanup

bdraco committed:
Include charging state for powerwall (#33432)

* Switch to binary sensor for charging status per review

* Powerwall charging margin of error is much higher than expected

balloob committed:
Add foundation for state translations (#34443)

balloob committed:
Fix return in translation script

Jc2k committed:
Remove deprecated homekit_controller credential storage locations (#34333)

* Remove deprecated homekit_controller credential storage locations

* async_import_legacy_pairing can also be removed

engrbm87 committed:
Add Islamic Prayer Times config flow (#31590)

* Add Islamic Prayer Times config_flow

* Add Islamic Prayer Times config_flow

* handle options update and fix tests

* fix sensor update handling

* fix pylint

* fix scheduled update and add test

* update test_init

* update flow options to show drop list

* clean up code

* async scheduling and revert state to timestamp

* fix update retry method

* update strings

* keep title as root key

elahd committed:
Add white value in light template platform (#32481)

* Added support for white value in Template Light integration.

* Tests now working.

* Clean up

* Make template more robust

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bieniu committed:
Bump brother to version 0.1.13 (#34456)

escoand committed:
Add config flow and device registry to fritzbox integration (#31240)

* add config flow

* fix pylint

* update lib

* Update config_flow.py

* remote devices layer in config

* add default host

* avoid double setups of entities

* remove async_setup_platform

* store entities in hass.data

* pass fritz connection together with config_entry

* fritz connections try no4 (or is it even more)

* fix comments

* add unloading

* fixed comments

* Update config_flow.py

* Update const.py

* Update config_flow.py

* Update __init__.py

* Update config_flow.py

* Update __init__.py

* Update __init__.py

* Update config_flow.py

* Update __init__.py

* Update __init__.py

* Update __init__.py

* Update config_flow.py

* add init tests

* test unloading

* add switch tests

* add sensor tests

* add climate tests

* test target temperature

* mock config to package

* comments

* test binary sensor state

* add config flow tests

* comments

* add missing tests

* minor

* remove string title

* deprecate yaml

* don't change yaml

* get devices async

* minor

* add devices again

* comments fixed

* unique_id fixes

* fix patches

* Fix schema

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

zhulik committed:
Improve the transmission integration (#34223)

* Update state after adding a new torrent

* Use cached torrents list in check_started_torrent_info

* Add torrent_info to all sensors

* Add torrent_info for active torrents

* Fix typo

* Update codeowners

* Do not set eta if it's unknown

* Fix codeowners

* Extract TransmissionSpeedSensor

* Extract TransmissionStatusSensor

* Extract TransmissionTorrentsSensor

* Refactor device_state_attributes() and update()

* Remove unused methods

* Use async_on_remove

* Fix sensor update

* Add transmission.remove_torrent service

* Add transmission_removed_torrent event

* Fix naming

* Fix typo in services.yaml

alistairg committed:
Add support for Lutron Keypad LEDs (#30452)

* Add support for Lutron Keypad LEDs

* Removed unneeded attribute definitions

* Pull initial state from Lutron on startup

* Format updates per code review

* Altered caching code to only fetch state if needed

* Update homeassistant/components/lutron/switch.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Cloud pylint is also offended by this ;)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

MrHarcombe committed:
- Added 'daily' mode as well as the original '3hourly'
- Added forecast data to the entity (not the sensor) integration
- Both sensor and entity integrations track the default position and reset their location if it changes
- Sensor and entity card icons are more meaningful, now

bdraco committed:
Add vacuum support to homekit (#34386)

fbradyirl committed:
Bump openwrt-luci-rpc version: 1.1.2 → 1.1.3   (#34463)

* Update manifest.json

* Update requirements_all.txt

Nossnevs committed:
Fix relative_time datetime object without timezone (#34273)

bdraco committed:
Switch griddy to async_on_remove (#34471)

bdraco committed:
Switch powerwall to async_on_remove (#34472)

bdraco committed:
Switch myq to async_on_remove (#34473)

bdraco committed:
Switch nexia to async_on_remove (#34474)

leroyshirto committed:
Update bt_smarthub component making it compatible with smarthub 2 (#31292)

* 0.2.0 of the btsmarthub_devicelist package makes it compatable with BT's home hub 2.
The API has changed in the new version so this change also makes the component code compatible with the changes to the library.

* Update homeassistant/components/bt_smarthub/device_tracker.py

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

* Update homeassistant/components/bt_smarthub/device_tracker.py

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

* Remove dep on config in BTSmartHubScanner

This should make BTSmartHubScanner easier to test as you can pass in a mock smarthub_client

* Black format bt_smarthub

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

bieniu committed:
Switch brother to async_on_remove (#34478)

pvizeli committed:
Bump hass-nabucasa 0.34.1 (#34480)

pvizeli committed:
Bump hass-nabucasa 0.34.1 (#34480)

balloob committed:
Import state strings (#34451)

balloob committed:
Update translations

balloob committed:
Migrate HomeKit to use describe_event for logbook support (#34485)

frenck committed:
Add transition support to scenes, cleanup blocking parameter (#34434)

Kane610 committed:
UniFi - Logs spam with not adding disabled entity (#34479)

* Proposed solution, unfortunately this makes the implementation less pretty

* Clarify why mac is registered in init

emontnemery committed:
Bump pychromecast (#34466)

* Bump pychromecast

* Fix tests

frenck committed:
Revert incorrect sync/async conversion in LCN integration (#34493)

Danielhiversen committed:
upgrade broadlink lib (#34488)

bieniu committed:
Switch airly to async_on_remove (#34498)

bieniu committed:
Switch gios to async_on_remove (#34499)

frenck committed:
Remove Pushetta integration (#34497)

dcnielsen90 committed:
Update bravia-tv backend (#34376)

* bravia-tv version bump

* Fix Auth Cookies

    - Fix authentication cookie expiration.
    - Remove obsolete error handling (since bravia-tv==1.0.2)
    - Remove obsolete mac address references. (since bravia-tv==1.0.2)
    - Update tests.

* fix async_refresh_playing_info

* Improve code format

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Config_entry needs mac address for HA restart

    This was mistakenly taken out. Home-Assistant needs to store mac address in
    case restarted. TV needs to be on to acquire mac address. This is neccesary
    to allow TV to be turned on by Home-Assistant after Home-Assistant is
    restarted.

* Skip update() during state changes

    It is best for performance to skip updates during state change because
    bravia.turn_on() will commonly push async_update() beyond it's update
    interval.

* update tests

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

pvizeli committed:
Fix cloud binary_sensor & TTS/STT (#34505)

pvizeli committed:
Fix cloud binary_sensor & TTS/STT (#34505)

frenck committed:
Remove Yahoo Weather integration (#34501)

MartinHjelmare committed:
Complete device registry update (#34500)

* Add manufacturer and model parameters to device registry update
  method.

springstan committed:
Drop UNIT_ prefix for constants (#34164)

* Drop UNIT_ prefix for VOLT

* Drop UNIT_ prefix for DEGREE

* Drop UNIT_ prefix for CONDUCTIVITY

* Drop UNIT_ prefix for UV_INDEX

* Run isort

ochlocracy committed:
Modify requirements for CameraStreamController in Alexa (#34470)

ochlocracy committed:
Modify requirements for CameraStreamController in Alexa (#34470)

bdraco committed:
Fix setting up multiple UPSes with NUT in 0.108.x (#34427)

ocalvo committed:
Fix a bug where a single long sms message is sent as multiple… (#34508)

Kane610 committed:
UniFi - Improve handling of client tracker is connected to better respect configured delay (#34510)

Kane610 committed:
UniFi - Only match ssid filter if ssid is present, this will only affect recently disconnected clients and clients which have a previously created entry in entity registry (#34509)

frenck committed:
Revert removal of JSON validator in hassfest (#34504)

Co-Authored-By: Paulus Schoutsen <balloob@gmail.com>

bdraco committed:
Display Homekit QR code when pairing (#34449)

* Display a QR code for homekit pairing

This will reduce the failure rate with HomeKit
pairing because there is less chance of entry
error.

* Add coverage

* Test that the qr code is created

* I cannot spell

* Update homeassistant/components/homekit/__init__.py

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

* Update homeassistant/components/homekit/__init__.py

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

ludeeus committed:
Rename translations dir for integrations (#34494)

balloob committed:
Bumped version to 0.108.7

emontnemery committed:
Don't log same MQTT message multiple times (#34511)

homeassistant committed:
[ci skip] Translation update

balloob committed:
Merge pull request #34517 from home-assistant/rc

bdraco committed:
Use registry to find linked batteries for homekit (#33519)

balloob committed:
Translation fixes and tweaks (#34489)

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

bdraco committed:
Resolve delays with tuya devices (#34512)

Kane610 committed:
UniFi - Simplify config option of block clients to just a multi select drop down (#34514)

bdraco committed:
Fix zeroconf interface being ineffective in HomeKit (#34516)

sdague committed:
Bump mychevy library to 2.0.1 (#34496)

ctalkington committed:
Improve Roku (#34431)

pvizeli committed:
Bump version 0.108.8

fredrike committed:
Improve volvooncall (#34495)

* Add write_ha_state to switch

* Add unique_id to voc

pvizeli committed:
Merge pull request #34527 from home-assistant/rc

0.108.8

bramkragten committed:
Updated frontend to 20200422.0 (#34532)

kennedyshead committed:
Bump aioasuswrt to 1.2.5 (#34454)

* bump aioasuswrt

Took 5 minutes

* bump aioasuswrt

Took 10 minutes

Co-authored-by: magnusknutas <magnus@thefarm.se>

balloob committed:
Don't set step title if it's integration title (#34524)

balloob committed:
Validate translations for custom components (#34519)

MatsNl committed:
Add Atag One thermostat integration (#32361)

* add atag integration

* ignore

* generated

* Update .gitignore

* requirements update

* update coveragerc

* Revert "update coveragerc"

* make entity_types more readable

* add DOMAIN to listener

* entity name

* Use DataUpdateCoordinator

* fix translations

* enable preset_modes

* fix water_heater

* update coveragerc

* remove scan_interval

Co-Authored-By: J. Nick Koston <nick@koston.org>

* Apply suggestions from code review

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* fixes review remarks

* fix flake8 errors

* ensure correct HVACmode

* add away mode

* use write_ha_state instead of refresh

* remove OFF support, add Fahrenheit

* rename test_config_flow.py

Co-authored-by: J. Nick Koston <nick@koston.org>
Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

frenck committed:
Merge branch 'master' into dev

jjlawren committed:
Deprecate Plex YAML config (#34546)

fronzbot committed:
Upgrade blinkpy library to 0.14.3 (#34542)

jlmcgehee21 committed:
Fix Bayesian Binary JSON Serialization (#34540)

Kane610 committed:
UniFi - Improve logging related to loosing connection to controller (#34547)

jrester committed:
Update tesla-powerwall to version 0.2.8 (#34545)

* Update tesla-powerwall to version 0.2.7

* Update tesla-powerwall to version 0.2.8

jjlawren committed:
Add All wrapper to deprecated Plex schema (#34552)

emontnemery committed:
Remove reconnect logic from MQTT client. (#34556)

Quentame committed:
Bump python-synology to 0.7.0 (#34534)

uphillbattle committed:
Set mqtt binary_sensor unavailable if expire_after specified (#34259)

* Set self._expired=True if expire_after specified

* Added test_expiration_on_discovery_and_discovery_update_of_binary_sensor to mqtt/test_binary_sensor.py

* Fixed flake8 error

* Fixed isort error

emontnemery committed:
Only subscribe when MQTT client is connected. (#34557)

bachya committed:
Add support for AirVisual Node/Pro units (#32815)

* Add support for AirVisual Node Pro units

* Fixed tests

* Updated dependencies

* Guard looks cleaner

* Limit options update to geography-based entries

* Docstring

* Use proper precision in display_temp

* Add availability for AirVisualNodeProSensor

* Updated translations

* Samba stuff in play

* Wrap up Samba

* Fix tests

* Remove unnecessary updates

* Normalize labels

* Bump requirements

* Don't include configuration.yaml support for this new functionality

* Fix tests

* Code review

* Code review

* Update coveragerc

* Code review

homeassistant committed:
[ci skip] Translation update

ziv1234 committed:
Add coverage to qwikswitch (#33939)

MatthewFlamm committed:
Fix deleting and readding nws entry (#34555)

* fix deleting and readding nws

* Clean up

* Fix variable name clash

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

MartinHjelmare committed:
Remove hap pyc file (#34563)

Kane610 committed:
UniFi - Store controller in config_entry.entry_id (#34553)

* Store controller in config_entry.entry_id

* Clean up imports

jrester committed:
Powerwall sensor add is_active, round state attributes and change thresholding for charging status sensor (#34582)

* Change sensor values and thresholding

* Update tests

bachya committed:
Handle flaky SimpliSafe notification registration (#34475)

bachya committed:
Add service to clear SimpliSafe notifications (#34481)

bdraco committed:
Remember homekit aids for entities without a unique id (#34587)

* Remember homekit aids for entities without a unique id

* add backwards compat

* increase cover

Kane610 committed:
UniFi - Move some preloading of unavailable clients earlier in setup phase (#34599)

Improve readability of setting up switch platform

frenck committed:
Upgrade pre-commit to 2.3.0 (#34604)

frenck committed:
Fix Garmin Connect i/o in event loop (#34598)

ludeeus committed:
Limit clone/view stats to repos with push access (#34575)

jjlawren committed:
Add Plex play_media logging and troubleshooting tools (#34412)

frenck committed:
Remove old style translations from Atag (#34585)

emontnemery committed:
Rename BinarySensorDevice to BinarySensorEntity (#34462)

* Rename BinarySensorDevice to BinarySensorEntity

* Tweak

* Move deprecation warning to __new__, add test

* Move deprecation warning back to __init__

* Move deprecation warning to __init_subclass

frenck committed:
Upgrade mock-open to 1.4.0 (#34606)

frenck committed:
Upgrade pyupgrade to v2.2.1 (#34608)

bdraco committed:
Restore ability to overwrite homekit max temp bound (#34612)

frenck committed:
Upgrade codecov to 2.0.22 (#34607)

starkillerOG committed:
Use "arming" state during transition in manual alarm panel (#32950)

* Manual Alarm Control Panel: use proper "Arming" state

* Update previous and next attributes

* add CONF_ARMING_TIME

* Split up arming and pending time, pending_time --> arming_time

* update tests

* fix issort

* fix issort

* fix demo platform

* fix alarm test

* remove arming_time from the triggered state

* Match previous default "delay_time"

* fix tests

* fix arming state when triggering

* fix arming _arming_time_by_state for Triggering state

* change to not in list

* Update homeassistant/components/manual/alarm_control_panel.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* async_update_ha_state -> async_write_ha_state

* black formatting

* add Callback

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* import callback

* update device triggers alarm_control_panel

* Update test_device_trigger.py

* Update device_trigger.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

balloob committed:
Fix UVC doing I/O in event loop (#34610)

fabaff committed:
Move myStrom light and switch to async (#34079)

frenck committed:
Fix BloomSky KeyError: 'monitored_conditions' (#34613)

Co-Authored-By: Paulus Schoutsen <balloob@gmail.com>

homeassistant committed:
[ci skip] Translation update

bdraco committed:
Handle synology_dsm discovery broadcasting on multiple ip addresses (#34623)

jjlawren committed:
Refactor Plex device/session updates (#34616)

jjlawren committed:
Remove deprecated cert_expiry config (#34628)

balloob committed:
Delay sync for Google and limit updates to relevant info (#34622)

MrHarcombe committed:
Refactored MetOfficeData out (as it's no longer just "current" data)
Location configuration is tracked if it's taken from the core

homeassistant commented:
Hi @MrHarcombe, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#34642).

Once you do that we will be able to review and accept this pull request.

Thanks!

MrHarcombe committed:
Merge branch 'dev' into metoffice_updates

MrHarcombe committed:
Fix to homeassisant.const imports

Suspect it must have been a merge defect, but the imports had got out of whack.

MrHarcombe committed:
Import corrections after failing various lint checks

MrHarcombe committed:
- Moved all the DataPoint logic into the MetOfficeData class
- Added test cases for both weather and sensor entities
- Having problems with testing the case after an event is triggered (any help gratefully received)

MartinHjelmare commented on homeassistant/components/metoffice/weather.py:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

If you want to change the configuration, please first make a separate PR where the integration is refactored to use config flow and config entries. When that is merged we can rebase here and continue.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

MrHarcombe commented on homeassistant/components/metoffice/weather.py:
Right-o. Can you point me to an example of how to do this for something like the weather integrations, please, as I've been struggling to do this over the last few days? All the documentation assumes a nice simple single entry-point entity, but of course the weather integrations are both sensor and entity, so they don't work quite the same.

I'm happy to start a new branch from the current "dev", without my changes on, to try and get this done first... but I could do with a working example of how to get the config_flow working for platform schema based projects as a template.

Any thoughts?

MartinHjelmare commented on homeassistant/components/metoffice/weather.py:
The nws integration has three different endpoints but just a weather platform. It's easy to add a sensor platform to that though. Just forward the entry to the sensor domain too, and then have the sensor platform entities subscribe to updates from the data update coordinator(s).

https://github.com/home-assistant/core/blob/1c84905bcd05d27d642f1e6a6a8c9ab553b45b2f/homeassistant/components/nws/__init__.py#L66-L111

MrHarcombe commented:
Excellent, thanks. I'll take a look in there and see what I can do. Watch
this space 😀

On Sun, 26 Apr 2020, 21:47 Martin Hjelmare, <notifications@github.com>
wrote:

> *@MartinHjelmare* commented on this pull request.
> ------------------------------
>
> In homeassistant/components/metoffice/weather.py
> <https://github.com/home-assistant/core/pull/34642#discussion_r415397380>:
>
> >  PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
>      {
>          vol.Required(CONF_API_KEY): cv.string,
>          vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
> +        vol.Optional(CONF_MODE, default=MODE_3HOURLY): vol.In(
>
> The nws integration has three different endpoints but just a weather
> platform. It's easy to add a sensor platform to that though. Just forward
> the entry to the sensor domain too, and then have the sensor platform
> entities subscribe to updates from the data update coordinator(s).
>
>
> https://github.com/home-assistant/core/blob/1c84905bcd05d27d642f1e6a6a8c9ab553b45b2f/homeassistant/components/nws/__init__.py#L66-L111
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/core/pull/34642#discussion_r415397380>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAQU6A2NC3MBSAB7BRXFBT3ROSMW7ANCNFSM4MQBUZQA>
> .
>

MrHarcombe commented:
Closed, will re-work and submit again following work done to move MetOffice integration to use config_flow.
